### ISSUE TITLE
AddBicyclePump, for bicycle_repair_station. Adding only pump

### ISSUE THREAD

**Issue by ArathonV** [streetcomplete/StreetComplete#5870]
**Use case**
A standalone bike pump is usually tagged as 
`service:bicycle:pump=yes
service:bicycle:tools=no `
the latter being assumed to be yes by default. However this currently seems impossible to tag in the app, as the pump can be added, and using the AddBicyclePump the pump tag can be set, the tools tag can not be set to no, thus incorrectly implying that this pump also provides tools.

**Proposed Solution**

- Add a Bike pump as an addable "thing"
- Add an "only" option to AddBicyclePump when asking about pumps that sets service:bicycle:tools=no

**Comment by matkoniecz** [comment]
> the latter being assumed to be yes by default.

why you think so?

> using the AddBicyclePump the pump tag can be set, the tools tag can not be set to no, thus incorrectly implying that this pump also provides tools.

well, in such case the same was also implied earlier

> Add a Bike pump as an addable "thing"

Iff it makes sense to have it also such way in iD then it can be added https://github.com/openstreetmap/id-tagging-schema

Or is it maybe already possible in iD?

(StreetComplete is using iD presets)

**Comment by ArathonV** [comment]
> why you think so?

the wiki says so, see [service:bicycle:tools](https://wiki.openstreetmap.org/wiki/Key:service:bicycle:tools)

**Comment by matkoniecz** [comment]
Hmm, in the worst case SC does not make things more clear - if 

- `amenity=bicycle_repair_station` was tagged without `service:bicycle:tools=no`
- `amenity=bicycle_repair_station` without `service:bicycle:tools=no` means that it has tools
- it has no tools

Then SC mapper adding `service:bicycle:pump=yes` still does not make any things worse.

**Comment by matkoniecz** [comment]
If new quest asking about `service:bicycle:tools=` makes sense (not sure!) then it seems that it was not proposed yet: https://github.com/search?q=repo%3Astreetcomplete%2FStreetComplete+service%3Abicycle%3Atools&type=issues

(see https://github.com/streetcomplete/StreetComplete/issues/new?assignees=&labels=&projects=&template=quest-suggestion.md - note that I have not checked is it fulfilling this requirements, I suspect that it fails "no spam" as vast majority of this stands has tools)

**Comment by ArathonV** [comment]
> Then SC mapper adding `service:bicycle:pump=yes` still does not make any things worse.

I agree, however I do feel like more, very useful, information could be easily added via the "only" suggestion.

>  I suspect that it fails "no spam" as vast majority of this stands has tools)

Hmm I am not so sure, in my experience, sole bike pumps are a bit more common, and currently around 40% are tagged with tools while almost 60% are tagged with pumps, but I guess that would be a discussion for another thread.

**Comment by mnalis** [comment]
> - Add a Bike pump as an addable "thing"

That wouldn't really help with already existing bicycle_repair_stations, though. (there is a Thing for `compressed air` which might work in some cases)

> - Add an "only" option to AddBicyclePump when asking about pumps that sets service:bicycle:tools=no

Might work, although IMHO that would be somewhat confusing, given the quest already asks only about _"Is there a working bicycle pump here?"_ (and not about tools, stands etc). Perhaps wording might be changed?

Some other options:
1. if it is rare enough (seems to depend on a country), user can already leave a note instead (using `Uh...` / `Can't say` / `Leave note`)
2. `Uh...` menu for that quest could be extended with `Only pump but no tools` which would tag `service:bicycle:pump=yes` + `service:bicycle:tools=no` , but that seems kludgy (and non-discoverable). Not much better than regular "only" answer suggested in previous comment.
3. extra quests `Are there bicycle tools available here?` (and `Is there a bicycle stand available here?`) could be made, but that might be spammy depending on the country (so might need research to only enable/disable it in some countries; which sounds like an awful lot of work).
4. The `AddBicyclePump` quest could be split into two -- one only about `shop=bicycle` (asking what this quest was doing), and another only about `amenity=bicycle_repair_station`. The latter quest in turn can be made in several ways:
    - 4.1 asking about `What is available on this bicycle repair station?`  and offering `pump`, `tools`, `both` (but that would miss the `stand` and other information, which is a shame if we're already extending the quest)
    - 4.2 asking about `Tag ALL equipment that is available on this bicycle repair station`, having multi-select (picture-based) options of `pump`, `stand`, `general tools`, `chain tool` (e.g. like recycling quest works, but also explicitly tag `=no` answer, not just `=yes`). That would provide most very useful information for cyclists, but makes it inconvenient for the mapper who then has to rummage through the tools to check if chain tool is available. 
    - 4.3 like `4.2`, but omit `chain tool` from the list to make it ***much*** easier to solve, at a loss of some useful information

Now, if this was [SCEE](https://wiki.openstreetmap.org/wiki/SCEE), I'd personally definitely go for `4.2`. 

However, This being StreetComplete and taking into consideration _"👨‍💻 Users are no experts"_ and _"🐿️ Easy answer"_ [quest guidelines](https://github.com/streetcomplete/StreetComplete/blob/master/QUEST_GUIDELINES.md), I'd probably go with `4.3` (although `4.1` + maybe `stands` half of `3.` as extra quest) would also fit.

What do people think?

**Comment by westnordost** [comment]
Bicycle pumps can be added to the things overlay as soon as there is a iD preset for that. For the iD preset, you'll have to open an issue or PR over there.

I like @mnalis suggestion to create a new quest that asks for **all** features of a bicycle repair station. Personally, I don't have such an issue with `chain_tool` or `stand` as long as representative photos or icons are found (chain tools always look quite similar).
The issue with these values is rather that `chain_tool` is rather specific. There are so many tools for bicycles, why single out that one tool? An incomplete list: Apart from  screwdrivers, allen wrenches and wrenches in different sizes, also tire levers, spoke tighteners or even patch kits, oil, chains, inner tubes. Sure, the latter are repair parts that would rather be sold at at least a vending machines, but for most repairs beyond tightening some screws or pumping your tires you need some repair parts, including for replacing your chain using a chain tool.

**Comment by mnalis** [comment]
**TL;DR:** make sense to me, we can also include `chain_tool` then to make a quest complete!

If other people would help find [representative and appropriately licensed photos](https://github.com/streetcomplete/StreetComplete/blob/master/CONTRIBUTING_A_NEW_QUEST.md#new-photos) for `pump` / `stand`/ `general tools` / `chain tool`; I can help with writing the quest itself as outlined in `4.2`.

---

<Details>
<summary>As for the `chain_tool`, here is rationale [click to expand]</summary>

> The issue with these values is rather that chain_tool is rather specific

As a mostly trekking/touring bicycle rider, that did strike me as odd too, but apparently (especially at MTB community) chain snapping is very frequent occurrence (to the point that at the courses here that is the _**second**_ thing they teach; after replacing the punctured tire).

>  why single out that one tool?

Probably because it is relatively often useful, but not always included (for this region I'd venture to guess that non-inclusion is because it is most likely to get stolen, as many bicycle users don't have it at home [as opposed to screwdrivers / wrenches] and, a specialty tool, it is more expensive then those).

From the OSM side, people found it useful enough to tag it several thousand times and document it on the [wiki](https://wiki.openstreetmap.org/wiki/Tag:amenity%3Dbicycle_repair_station) separately, and to single it out at [11.57%](https://taginfo.openstreetmap.org/keys/service%3Abicycle%3Achain_tool#combinations) of the places with general tools, so that sounds good enough for me for inclusion (even if I'm not a target user)

> Apart from screwdrivers, allen wrenches and wrenches in different sizes, also tire levers, spoke tighteners

The screewdrivers, wrenches, and tire levers are included in all service stations I've seen... Spoke tighteners often aren't, but probably those are not field-critical -- you can get home with slightly untrue wheel, but you can't without a chain or with a flat tire. Also, while truing the wheel is possible without special stand, it is not easy, require significant expertise, and is very time consuming (and often might require few new spokes).  

> for most repairs beyond tightening some screws or pumping your tires you need some repair parts, 

I agree, but I've never seen replacement parts being given for free. In tourist heavy places, there might (very rarely) be an additional `amenity=vending_machine` + `vending=bicycle_tube` + `fee=yes` next to that `amenity=bicycle_repair_station` (or, more likely IME, nearby `shop=bicycle`+`service:bicycle:repair=yes`). But those would be always be tagged as separate objects...

> including for replacing your chain using the mentioned chain tool.

Actually (unless you run pully-less system), you can often fix a chain by shortening it for a link or two (or even more if you give up shifting speeds). Even for my pully-less integrated Shimano Nexus shifter (using single-speed chain), I could survive chain snapping and resulting single link loss (not that it had ever happened to me, but than again those single-speed chains are robust monsters compared to even 10/11-speeds ones, not to mention those thin 13-speed) .

</details>

**Comment by westnordost** [comment]
So, anyway, would you create a new issue in the form of a "new quest suggestion" for this, please?

**Comment by mnalis** [comment]
> For such a quest, I still would find it meaningful to first ask in the community forum if there are other popular or sought-after features of a bicycle repair station should be selectable.

Asked at https://community.openstreetmap.org/t/useful-subtags-for-amenity-bicycle-repair-station-for-streetcomplete-quest/118491

**Comment by ArathonV** [comment]
As sugested I have opened an issue for this preset here: https://github.com/openstreetmap/id-tagging-schema/issues/1339

**Comment by mnalis** [comment]
>  Asked at https://community.openstreetmap.org/t/useful-subtags-for-amenity-bicycle-repair-station-for-streetcomplete-quest/118491

The results AFAICT seem to be in favor of current selection of `pump`/`tools`/`chain_tool`/`stand`. 

There were some mentions of few other undocumented tags in use (like `bicycle:service:screwdriver`, `bicycle:service:wrench`, `bicycle:service:tire_lever` etc.), but those were determined to have low usages and are (in majority of the cases) implied by `bicycle:service:tools=yes`. 

It was also mentioned that such `amenity=bicycle_repair_station` are sometimes accompanied by `amenity=charging_station`, but that this is was better mapped as separate node.

The discussion then veered in direction of separate valves  (and Schwalbe Clik valve in particular) quest (to be asked if `bicycle:service:pump=yes`) ; and as a result that valves quest has since been implemented for SCEE (as deemed [too technical for SC](https://github.com/streetcomplete/StreetComplete/issues/4450)) in https://github.com/Helium314/SCEE/pull/644.

> So, anyway, would you create a new issue in the form of a "new quest suggestion" for this, please?

(I'll open one shortly and reference it here.)

**Comment by westnordost** [comment]
Alright, would you pour this information into a new quest suggestion? Then this issue can be closed.

**Comment by mnalis** [comment]
(Ok, created new quest suggestion with template in https://github.com/streetcomplete/StreetComplete/issues/5910, closing this issue as duplicate...)
